wvd-templates/Update existing WVD host pool/mainTemplate.json

```diff
@@ -385,65 +385,61 @@
-                            "password": "PrivateSettingsRef:tenantAdminPassword"
-                        },
-                        "ADAdminCredentials": {
-                            "userName": "[parameters('existingDomainUPN')]",
-                            "password": "PrivateSettingsRef:adAdminPassword"
-                        },
                         "RDBrokerURL": "[parameters('rdBrokerURL')]",
                         "DefinedTenantGroupName": "[parameters('existingTenantGroupName')]",
                         "TenantName": "[variables('existingTenantName')]",
                         "HostPoolName": "[parameters('existingHostpoolName')]",
                         "Hours": "[variables('registrationExpirationHours')]",
+                        "TenantAdminCredentials": {
+                            "userName": "[parameters('tenantAdminUpnOrApplicationId')]",
+                            "password": "PrivateSettingsRef:tenantAdminPassword"
+                        },
                         "isServicePrincipal": "[parameters('isServicePrincipal')]",
-                        "AadTenantId": "[parameters('aadTenantId')]",
-                        "SubscriptionId": "[variables('subscriptionId')]",
-                        "DomainName": "[parameters('domainToJoin')]",
-                        "deleteordeallocateVMs": "[parameters('ActionOnPreviousVirtualMachines')]",
-                        "messageTitle": "[variables('messageTitle')]",
-                        "userNotificationMessege": "[parameters('userNotificationMessage')]",
-                        "userLogoffDelayInMinutes": "[parameters('userLogoffDelayInMinutes')]"
+                        "AadTenantId": "[parameters('aadTenantId')]"
                     }
                 },
                 "ProtectedSettings": {
                     "items": {
-                        "tenantAdminPassword": "[parameters('tenantAdminPassword')]",
-                        "adAdminPassword": "[parameters('existingDomainPassword')]"
+                        "tenantAdminPassword": "[parameters('tenantAdminPassword')]"
                     }
                 }
             }
         },
         {
-            "condition": "[greater(parameters('rdshNumberOfInstances'),1)]",
+            "condition": "[greater(parameters('rdshNumberOfInstances'), 2)]",
             "apiVersion": "2018-10-01",
             "type": "Microsoft.Compute/virtualMachines/extensions",
-            "name": "[concat(variables('rdshPrefix'), copyindex(1),'/', 'dscextension')]",
+            "name": "[concat(variables('rdshPrefix'), copyindex(2), '/', 'dscextension')]",
             "location": "[parameters('location')]",
             "dependsOn": [
-                "[concat('Microsoft.Compute/virtualMachines/', variables('rdshPrefix'),'0/extensions/dscextension')]"
+                "rdsh-domain-join-loop",
+                // note: if the following conditional dependencies are not deployed, arm will auto remove them
```

Comment:
I think we should clarify that this is talking about ARM removing it from the required dependencies as opposed to deleting some existing resources. Maybe something like:
If the following conditional dependencies are not deployed ARM will auto remove them from this dependency list.